### PR TITLE
OCT-2122: Layout Top Bar E2E & OCT-2137: Layout Footer E2E

### PR DESCRIPTION
**Pull request by jmikolajczyk**
## Definition of Done

1. [ ] If required, the desciption of your change is added to the [QA changelog](https://www.notion.so/octantapp/Changelog-for-the-QA-d96fa3b411cf488bb1d8d9a598d88281) 
2. [ ] Acceptance criteria are met.
3. [ ] PR is manually tested before the merge by developer(s).
    - [ ] Happy path is manually checked.
4. [ ] PR is manually tested by QA when their assistance is required (1).
    - [ ] Octant Areas & Test Cases are checked for impact and updated if required (2).
5. [ ] Unit tests are added unless there is a reason to omit them.
6. [ ] Automated tests are added when required.
7. [ ] The code is merged.
8. [ ] Tech documentation is added / updated, reviewed and approved (including mandatory approval by a code owner, should such exist for changed files).
    - [ ] BE: Swagger documentation is updated.
9. [ ] When required by QA:
    - [ ] Deployed to the relevant environment.
    - [ ] Passed system tests.

---

(1) Developer(s) in coordination with QA decide whether it's required. For small tickets introducing small changes QA assistance is most probably not required.

(2) [Octant Areas & Test Cases](https://docs.google.com/spreadsheets/d/1cRe6dxuKJV3a4ZskAwWEPvrFkQm6rEfyUCYwLTYw_Cc).

### CODE REVIEW

File: client/cypress/e2e/_4layoutTopBar.cy.ts [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+      });
+    }
+
+    describe('[TESTNET]', () => {
+      beforeEach(() => {
+        cy.window().then(win => {
+          // eslint-disable-next-line no-param-reassign
+          win.isTestnetCypress = true;
+          // wait for re-render
+          cy.wait(2500);
```
Comment (aziolek) [inline client/cypress/e2e/_4layoutTopBar.cy.ts:186]: Maybe we can wait for something more specific? Force relaod and wait for apploader to disappear?

File: client/cypress/support/index.ts [hunk 2]
```diff
@@ -1 +1,2 @@
+// import 'cypress-real-events';
```
Comment (aziolek) [inline client/cypress/support/index.ts:1]: To remove?

File: client/cypress/tsconfig.json [hunk 3]
```diff
@@ -8,9 +8,9 @@
       "@synthetixio/synpress/support",
       "cypress-wait-until",
       "@testing-library/cypress",
-      "../src/global"
+      "../global"
     ],
     "outDir": "./output"
   },
-  "include": ["**/*.*", "../synpress.config.ts"]
+  "include": ["**/*.*", "../synpress.config.ts", "cypress-real-events"]
```
Comment (aziolek) [inline client/cypress/tsconfig.json:15]: To remove?

File: client/yarn.lock [hunk 4]
```diff
@@ -15747,4 +15747,4 @@ zustand@^4.5.2:
 zwitch@^2.0.0:
   version "2.0.4"
   resolved "https://registry.yarnpkg.com/zwitch/-/zwitch-2.0.4.tgz#c827d4b0acb76fc3e685a4c6ec2902d51070e9d7"
-  integrity sha512-bXE4cR/kVZhKZX/RjPEflHaKVhUVl85noU3v6b8apfQEc1x4A+zBxjZ4lN8LqGd6WZ3dl98pY4o717VFmoPp+A==
+  integrity sha512-bXE4cR/kVZhKZX/RjPEflHaKVhUVl85noU3v6b8apfQEc1x4A+zBxjZ4lN8LqGd6WZ3dl98pY4o717VFmoPp+A==
```
Comment (aziolek) [inline client/yarn.lock:15750]: `yarn.lock` should not have been changed if no changes to packages were introduced. New line at the end of the default ending.

**Review by aziolek — CHANGES_REQUESTED** [review]

File: client/cypress/e2e/_4layoutTopBar.cy.ts (hunk 1 above)
Comment (jmikolajczyk) [reply client/cypress/e2e/_4layoutTopBar.cy.ts:186]: After reload `win.isTestnetCypress` will be undefined (it's not a value set from LS -> only for tests)

File: client/yarn.lock (hunk 4 above)
Comment (jmikolajczyk) [reply client/yarn.lock:15750]: There is no difference (hash is the same)

File: backend/tests/conftest.py [hunk 5]
```diff
@@ -656,18 +656,12 @@ def get_rewards_budget(self, address: str, epoch: int):
...
+        current_app.logger.debug("get_user_rewards_in_upcoming_epoch :", rv.text)
         return json.loads(rv.text)
 
     def get_user_rewards_in_epoch(self, address: str, epoch: int):
         rv = self._flask_client.get(f"/rewards/budget/{address}/epoch/{epoch}")
-        current_app.logger.debug(
-            "get_rewards_budget :",
-            self._flask_client.get(f"/rewards/budget/{address}/epoch/{epoch}").request,
-        )
+        current_app.logger.debug("get_rewards_budget :", rv.text)
```
[the remark this answers is not in the document]
Comment (jmikolajczyk) [reply backend/tests/conftest.py]: Reverted

File: backend/tests/conftest.py [hunk 6]
```diff
@@ -656,18 +656,12 @@ def get_rewards_budget(self, address: str, epoch: int):
 
     def get_user_rewards_in_upcoming_epoch(self, address: str):
         rv = self._flask_client.get(f"/rewards/budget/{address}/upcoming")
-        current_app.logger.debug(
-            "get_user_rewards_in_upcoming_epoch :",
-            self._flask_client.get(f"/rewards/budget/{address}/upcoming").request,
-        )
+        current_app.logger.debug("get_user_rewards_in_upcoming_epoch :", rv.text)
```
[the remark this answers is not in the document]
Comment (jmikolajczyk) [reply backend/tests/conftest.py]: Reverted

File: backend/app/infrastructure/external_api/safe/message_details.py [hunk 7]
```diff
@@ -1,3 +1,4 @@
+from http import HTTPStatus
```
[the remark this answers is not in the document]
Comment (jmikolajczyk) [reply backend/app/infrastructure/external_api/safe/message_details.py]: Reverted

File: backend/app/infrastructure/external_api/safe/message_details.py [hunk 8]
```diff
@@ -13,7 +14,8 @@ def get_message_details(message_hash: str, is_mainnet: bool) -> dict:
         response.raise_for_status()
         json_response = response.json()
     except requests.exceptions.RequestException as e:
-        app_module.ExceptionHandler.print_stacktrace(e)
+        if response.status_code != HTTPStatus.NOT_FOUND:
+            app_module.ExceptionHandler.print_stacktrace(e)
```
[the remark this answers is not in the document]
Comment (jmikolajczyk) [reply backend/app/infrastructure/external_api/safe/message_details.py]: Reverted

File: client/yarn.lock (hunk 4 above)
Comment (aziolek) [reply client/yarn.lock:15750]: I know.

**Review by aziolek — APPROVED** [review]

File: client/yarn.lock (hunk 4 above)
Comment (aziolek) [reply client/yarn.lock:15750]: It's still here.

**Review by aziolek — APPROVED** [review]

**Pull request merged**